### PR TITLE
Hotfix/sg 000 transform backward compatibility

### PR DESCRIPTION
**Pull request by Louis-Dupont**
Note: still need to update the tests

### CODE REVIEW

File: src/super_gradients/training/transforms/transforms.py [hunk 1]
```diff
@@ -775,20 +775,43 @@ class DetectionTargetsFormatTransform(DetectionTransform):
...
+        input_dim: Optional[tuple] = None,
         input_format: ConcatenatedTensorFormat = XYXY_LABEL,
         output_format: ConcatenatedTensorFormat = LABEL_CXCYWH,
         min_bbox_edge_size: float = 1,
         max_targets: int = 120,
     ):
         super(DetectionTargetsFormatTransform, self).__init__()
+        if isinstance(input_format, DetectionTargetsFormat) or isinstance(output_format, DetectionTargetsFormat):
+            raise TypeError(
+                "DetectionTargetsFormat is not supported for input_format and output_format starting from super_gradients==3.0.7.\n"
```
Comment (shaydeci) [inline src/super_gradients/training/transforms/transforms.py:787]: I don't get this...

**Review by shaydeci — CHANGES_REQUESTED** [review]
one clarification inline.

File: src/super_gradients/training/transforms/transforms.py (hunk 1 above)
Comment (Louis-Dupont) [reply src/super_gradients/training/transforms/transforms.py:787]: I raise an error if they still use DetectionTargetsFormat (the Enum that we used before). It's to make the error explicit for a few versions.
Without this the error that they get if they dont change their old recipe is hard to understand.

**Review by shaydeci — APPROVED** [review]
LGTM

**Pull request merged**